### DISCUSSION TITLE
Regarding Payment Gateway Redirect Methods

### DISCUSSION THREAD

**Discussion by knit-pay** [pronamic/wp-pay-core#48] [General]
Hello Team

I understand, currently, there are 2 types of payment gateway redirect methods, `METHOD_HTTP_REDIRECT` and `METHOD_HTML_FORM`. In `METHOD_HTTP_REDIRECT`, the customer gets redirected to the action URL directly and in `METHOD_HTML_FORM`, a redirect page is visible to the customer (with Pay Now button) before he gets redirected to the payment gateway page.

I understand that in the `METHOD_HTML_FORM` redirection method, we can pass custom output fields on the payment redirect page using the `get_output_fields` function, which is getting displayed using the `get_form_html` function.

I want to know if there is a way to do `METHOD_HTML_FORM` redirection without showing the redirect page to the customer? For example on the WooCommerce checkout page, after selecting a payment method and clicking "Pay Now", the customer should directly redirect to the payment gateway page, instead of showing the redirection page. Can we show a hidden custom field on the WooCommerce checkout page, instead of a redirection page and redirect customers directly to the payment gateway page.
This particular thing will help me several ways. For example, there are certain payment gateways, in which both `METHOD_HTTP_REDIRECT` and `METHOD_HTML_FORM` don't work. In these payment gateways, they show the iFrame popup box directly on the checkout page instead of any kind of redirection. As a workaround, I am overriding the `get_form_html` function and doing all the javascript-related tasks on the redirection page, and showing the iFrame pop-up box on the redirection page. But if we will be able to load `get_form_html` on the checkout page instead of the redirection page, it will look good for particularly those payment gateways in which there is no redirection at all.

What do you suggest, is there any functionality already there to handle iframe related checkouts? Is there any better way to implement it? Can we do the enhancement and show the output from `get_form_html` function directly in extensions checkout instead of redirect page?

**Comment by rvdsteege** [comment]
> What do you suggest, is there any functionality already there to handle iframe related checkouts?

There is no specific functionality for this.

> Can we show a hidden custom field on the WooCommerce checkout page

Depending on what this field should contain, you might be able to do something with `get_input_fields()` or `print_fields()` in https://github.com/wp-pay-extensions/woocommerce/blob/develop/src/Gateway.php#L1055-L1073

**Comment by knit-pay** [comment]
Thanks for your suggestions. I will look into it and will let you know if I need additional help.
